torchgeo/datasets/openbuildings.py

```diff
@@ -0,0 +1,464 @@
+
+        if not filepaths:
+            raise IndexError(
+                f"query: {query} not found in index with bounds: {self.bounds}"
+            )
+
+        shapes = self._filter_geometries(query, filepaths)
+
+        # Rasterize geometries
+        width = (query.maxx - query.minx) / self.res
+        height = (query.maxy - query.miny) / self.res
+        transform = rasterio.transform.from_bounds(
+            query.minx, query.miny, query.maxx, query.maxy, width, height
+        )
+        if shapes:
+            masks = rasterio.features.rasterize(
+                shapes, out_shape=(int(height), int(width)), transform=transform
+            )
+            masks = torch.tensor(masks).unsqueeze(0)  # type: ignore[attr-defined]
+        else:
+            masks = torch.zeros(  # type: ignore[attr-defined]
+                size=(1, int(height), int(width))
+            )
+
+        sample = {"mask": masks, "crs": self.crs, "bbox": query}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _filter_geometries(
+        self, query: BoundingBox, filepaths: List[str]
+    ) -> List[Dict[str, Any]]:
+        """Filters a df read from the polygon csv file based on query and conf thresh.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+            filepaths: filepaths to files that were hits from rmtree index
+
+        Returns:
+            List with all polygons from all hit filepaths
+
+        """
+        # We need to know the bounding box of the query in the source CRS
+        (minx, maxx), (miny, maxy) = fiona.transform.transform(
+            self._crs.to_dict(),
+            self._source_crs.to_dict(),
+            [query.minx, query.maxx],
+            [query.miny, query.maxy],
+        )
+        df_query = (
+            "longitude >= {} & longitude <= {} & " "latitude >= {} & latitude <= {}"
+        ).format(minx, maxx, miny, maxy)
+        shapes = []
+        for f in filepaths:
+            csv_chunks = pd.read_csv(f, chunksize=200000, compression="gzip")
+            for chunk in csv_chunks:
+                df = chunk.query(df_query)
+                # Warp geometries to requested CRS
+                polygon_series = df["geometry"].map(self._wkt_fiona_geom_transform)
+                shapes.extend(polygon_series.values.tolist())
+
+        return shapes
+
+    def _wkt_fiona_geom_transform(self, x: str) -> Any:
+        """Function to transform a geometry string into new crs.
+
+        Args:
+            x: Polygon string
+
+        Returns:
+            transformed geometry in geojson format
+
+        """
+        x = json.dumps(shapely.geometry.mapping(wkt.loads(x)))
+        x = json.loads(x.replace("'", '"'))
+        return fiona.transform.transform_geom(
+            self._source_crs.to_dict(), self.crs.to_dict(), x
+        )
+
+    def _verify(self) -> None:
```

Comment:
We should probably also verify that a `tiles.geojson` file exists

Reply:
I think I am verifying that in line 408.

Reply:
Ah yes, sorry, missed that.